### DISCUSSION TITLE
Copilot must tell us when it loses context of the conversations

### DISCUSSION THREAD

**Discussion by drastawi** [community/community#109897] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Often times I have very productive conversations and it is clear that at some point the Copilot forgets everything. It can often remember details from long log output a 10 messages before but too often as some point it completely forgets everything we were talking about and starts to hallucinate about important pieces of data like IPs.

If I really drill down into it, it always ends like this:
> Sorry, the last message was not related, could you send me the curl command you told me about one message before?

I get the answer 
> I'm sorry, but as an AI, I don't have the ability to recall previous conversations or messages...

It is very clear to me the Copilot knows when it lost the memory, but it really refuses to mention there is any problem and proceeds to waste a lot of my time. As soon as it loses the memory it has to say "I forgot the IP", "I forgot the command", "I forgot the point of this conversation" or just disconnect. This is such a huge waste of time but I can't imagine this is hard to fix.

**Comment by ChesedPrather** [comment]
It's designed to be stateful and maintain the context of the **session**. When you see a message such as _"I don't have the ability to recall previous conversations"_ that is a true statement. But it is a bug to say _"or messages"_.  The fact that it cannot recall previous messages of the **current conversation** makes Copilot barely a beta not ready for primetime.  Certainly not a paid subscription.

**Reply by jacques** [reply]
The loosing chat context on the second or third message is quite annoying as you are starting to get into the conversation to define the requirements of what you are working on.

**Comment by nodered** [comment]
I found some MCP solutions for general memory and agent memory that may be useful... https://github.com/punkpeye/awesome-mcp-servers?tab=readme-ov-file#-knowledge--memory 

I haven't tried these yet, I just stumbled onto MCP recently.

Is this out of scope for Github Copilot? If it is out of scope, it opens up some possibilities for Chat Participants!
